likecoin/admin/restful.php

```diff
@@ -45,6 +45,181 @@ function likecoin_rest_refresh_publish_status( $request ) {
 	$data         = likecoin_parse_publish_status( $matters_info );
 	return new WP_REST_Response( $data, 200 );
 }
+/**
+ * Add refresh publish status endpoint
+ *
+ * @param object| $post WordPress post object.
+ */
+function likecoin_get_post_image_url( $post ) {
+	$urls                  = array();
+	$content               = apply_filters( 'the_content', $post->post_content );
+	$dom_document          = new DOMDocument();
+	$libxml_previous_state = libxml_use_internal_errors( true );
+	$dom_content           = $dom_document->loadHTML( '<template>' . mb_convert_encoding( $content, 'HTML-ENTITIES' ) . '</template>', LIBXML_HTML_NOIMPLIED | LIBXML_HTML_NODEFDTD );
+	$images                = $dom_document->getElementsByTagName( 'img' );
+
+	// get all images.
+	foreach ( $images as $image ) { // only works after attachment is converted to image by user.
+		// phpcs:disable WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+		$parent = $image->parentNode;
+		if ( 'figure' === $parent->nodeName ) {
+			// phpcs:enable WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+			$classes = $parent->getAttribute( 'class' );
+			$parent->setAttribute( 'class', $classes . ' image' );
+		} else {
+			$figure = $dom_document->createElement( 'figure' );
+			$figure->setAttribute( 'class', 'image' );
+			$image = $parent->replaceChild( $figure, $image );
+			$figure->appendChild( $image );
+			$parent = $figure;
+		}
```

Comment:
I think this whole part is not needed if you only want the img src url

Reply:
True, deleted.